lib/http1.c

```diff
@@ -586,9 +648,18 @@ static void cleanup_connection(struct st_h2o_http1_conn_t *conn)
         return;
     }
 
+    if (conn->req.proceed_req) {
+        conn->_req_entity_reader = NULL;
+        set_timeout(conn, 0, NULL);
+        h2o_socket_read_stop(conn->sock);
```

Comment:
I'm not sure if this is the right moment. Don't we need to disarm the timeout and stop reading additional bytes every time when we pass something to the handler? Otherwise, the amount of data we buffer cannot be capped.

Reply:
Addressed in e4d894b

Reply:
e4d894b looks good to me, but would you mind elaborating why we need to stop reading  (or consider the case of `conn->req.proceed_req` being non-NULL) in this function, after consulting the value of `conn->req.http1_is_persistent`?

It is my understanding that `cleanup_connection` is called only when the HTTP/1 stack successfully processes a request and sending a response. Assuming that is still the case, I _think_ `proceed_req` and `_req_entity_reader` must have been set to NULL, and reading from the socket should have been stopped. Maybe what we need are assertions.

Reply:
I've added assertions, thanks for the suggestion.

Reply:
> I've added assertions, thanks for the suggestion.

And the assertions found that 603c731 is necessary because `h2o_send_error` can complete the request without closing the connection.